### PR TITLE
[glpk] Fix feature build error

### LINKED ISSUE #35244: glpk[core,mysql]:x64-windows Build failure
Package: glpk[core,dl,mysql]:x64-windows -> 5.0#2

**Host Environment**

- Host: x64-windows
- Compiler: MSVC 19.37.32825.0
-    vcpkg-tool version: 2999-12-31-unknownhash
    vcpkg-scripts version: ecf95e1e7 2023-11-18 (4 days ago)

**To Reproduce**

`vcpkg x-test-features --ci-feature-baseline=scripts/ci.feature.baseline.txt --failure-logs=log --all`
**Failure logs**

```
-- Using cached glpk.tar.gz.
-- Extracting source G:/v/downloads/glpk.tar.gz
-- Applying patch configure.ac.patch
-- Using source at G:/v/b1/glpk/src/glpk-de479f8ea9.clean
-- Found external ninja('1.11.0').
-- Getting CMake variables for x64-windows
-- Using cached msys2-autoconf-wrapper-20221207-1-any.pkg.tar.zst.
-- Using cached msys2-automake-wrapper-20221207-1-any.pkg.tar.zst.
-- Using cached msys2-binutils-2.40-1-x86_64.pkg.tar.zst.
-- Using cached msys2-libtool-2.4.7-3-x86_64.pkg.tar.zst.
-- Using cached msys2-make-4.4.1-1-x86_64.pkg.tar.zst.
-- Using cached msys2-pkgconf-1.9.4-1-x86_64.pkg.tar.zst.
-- Using cached msys2-which-2.21-4-x86_64.pkg.tar.zst.
-- Using cached msys2-bash-5.2.015-1-x86_64.pkg.tar.zst.
-- Using cached msys2-coreutils-8.32-5-x86_64.pkg.tar.zst.
-- Using cached msys2-file-5.44-5-x86_64.pkg.tar.zst.
-- Using cached msys2-gawk-5.2.1-2-x86_64.pkg.tar.zst.
-- Using cached msys2-grep-1~3.0-6-x86_64.pkg.tar.zst.
-- Using cached msys2-gzip-1.12-2-x86_64.pkg.tar.zst.
-- Using cached msys2-diffutils-3.9-1-x86_64.pkg.tar.zst.
-- Using cached msys2-sed-4.9-1-x86_64.pkg.tar.zst.
-- Using cached msys2-msys2-runtime-3.4.6-1-x86_64.pkg.tar.zst.
-- Using cached msys2-autoconf2.71-2.71-1-any.pkg.tar.zst.
-- Using cached msys2-automake1.16-1.16.5-1-any.pkg.tar.zst.
-- Using cached msys2-libiconv-1.17-1-x86_64.pkg.tar.zst.
-- Using cached msys2-libintl-0.21-2-x86_64.pkg.tar.zst.
-- Using cached msys2-zlib-1.2.13-1-x86_64.pkg.tar.zst.
-- Using cached msys2-tar-1.34-3-x86_64.pkg.tar.zst.
-- Using cached msys2-findutils-4.9.0-3-x86_64.pkg.tar.zst.
-- Using cached msys2-gmp-6.2.1-2-x86_64.pkg.tar.zst.
-- Using cached msys2-gcc-libs-11.3.0-3-x86_64.pkg.tar.zst.
-- Using cached msys2-libbz2-1.0.8-4-x86_64.pkg.tar.zst.
-- Using cached msys2-liblzma-5.4.1-1-x86_64.pkg.tar.zst.
-- Using cached msys2-libzstd-1.5.4-1-x86_64.pkg.tar.zst.
-- Using cached msys2-libreadline-8.2.001-3-x86_64.pkg.tar.zst.
-- Using cached msys2-mpfr-4.2.0-2-x86_64.pkg.tar.zst.
-- Using cached msys2-libpcre-8.45-3-x86_64.pkg.tar.zst.
-- Using cached msys2-m4-1.4.19-2-x86_64.pkg.tar.zst.
-- Using cached msys2-perl-5.36.0-1-x86_64.pkg.tar.zst.
-- Using cached msys2-ncurses-6.4-1-x86_64.pkg.tar.zst.
-- Using cached msys2-libcrypt-2.1-4-x86_64.pkg.tar.zst.
-- Using msys root at G:/v/downloads/tools/msys2/d7266db249278763
-- Generating configure for x64-windows
-- Finished generating configure for x64-windows
-- Using cached mingw-w64-i686-pkgconf-1~1.8.0-2-any.pkg.tar.zst.
-- Using cached msys2-msys2-runtime-3.4.6-1-x86_64.pkg.tar.zst.
-- Using msys root at G:/v/downloads/tools/msys2/6f3fa1a12ef85a6f
-- Configuring x64-windows-dbg
-- Configuring x64-windows-rel
-- Building x64-windows-dbg
-- Installing x64-windows-dbg
-- Building x64-windows-rel
-- Installing x64-windows-rel
-- Fixing pkgconfig file: G:/v/vcpkg1/packages/glpk_x64-windows/lib/pkgconfig/glpk.pc
CMake Error at scripts/cmake/vcpkg_fixup_pkgconfig.cmake:141 (message):
  G:/v/downloads/tools/msys2/6f3fa1a12ef85a6f/mingw32/bin/pkg-config.exe
  --exists glpk failed with error code: 1

      ENV{PKG_CONFIG_PATH}: "G:/v/vcpkg1/packages/glpk_x64-windows/lib/pkgconfig;G:/v/vcpkg1/packages/glpk_x64-windows/share/pkgconfig;G:/v/vcpkg1/vcpkg_installed/x64-windows/lib/pkgconfig;G:/v/vcpkg1/vcpkg_installed/x64-windows/share/pkgconfig"
      output: Package mysql was not found in the pkg-config search path.

  Perhaps you should add the directory containing `mysql.pc'

  to the PKG_CONFIG_PATH environment variable

  Package 'mysql', required by 'glpk', not found
Call Stack (most recent call first):
  scripts/cmake/vcpkg_fixup_pkgconfig.cmake:206 (z_vcpkg_fixup_pkgconfig_check_files)
  ports/glpk/portfile.cmake:64 (vcpkg_fixup_pkgconfig)
  scripts/ports.cmake:170 (include)



```

### LINKED ISSUE #35245: glpk[core,odbc]:x64-windows Build failure
Package: glpk[core,dl,odbc]:x64-windows -> 5.0#2

**Host Environment**

- Host: x64-windows
- Compiler: MSVC 19.37.32825.0
-    vcpkg-tool version: 2999-12-31-unknownhash
    vcpkg-scripts version: ecf95e1e7 2023-11-18 (4 days ago)

**To Reproduce**

`vcpkg x-test-features --ci-feature-baseline=scripts/ci.feature.baseline.txt --failure-logs=log --all`
**Failure logs**

```
-- Using cached glpk.tar.gz.
-- Extracting source G:/v/downloads/glpk.tar.gz
-- Applying patch configure.ac.patch
-- Using source at G:/v/b1/glpk/src/glpk-de479f8ea9.clean
-- Found external ninja('1.11.0').
-- Getting CMake variables for x64-windows
-- Using cached msys2-autoconf-wrapper-20221207-1-any.pkg.tar.zst.
-- Using cached msys2-automake-wrapper-20221207-1-any.pkg.tar.zst.
-- Using cached msys2-binutils-2.40-1-x86_64.pkg.tar.zst.
-- Using cached msys2-libtool-2.4.7-3-x86_64.pkg.tar.zst.
-- Using cached msys2-make-4.4.1-1-x86_64.pkg.tar.zst.
-- Using cached msys2-pkgconf-1.9.4-1-x86_64.pkg.tar.zst.
-- Using cached msys2-which-2.21-4-x86_64.pkg.tar.zst.
-- Using cached msys2-bash-5.2.015-1-x86_64.pkg.tar.zst.
-- Using cached msys2-coreutils-8.32-5-x86_64.pkg.tar.zst.
-- Using cached msys2-file-5.44-5-x86_64.pkg.tar.zst.
-- Using cached msys2-gawk-5.2.1-2-x86_64.pkg.tar.zst.
-- Using cached msys2-grep-1~3.0-6-x86_64.pkg.tar.zst.
-- Using cached msys2-gzip-1.12-2-x86_64.pkg.tar.zst.
-- Using cached msys2-diffutils-3.9-1-x86_64.pkg.tar.zst.
-- Using cached msys2-sed-4.9-1-x86_64.pkg.tar.zst.
-- Using cached msys2-msys2-runtime-3.4.6-1-x86_64.pkg.tar.zst.
-- Using cached msys2-autoconf2.71-2.71-1-any.pkg.tar.zst.
-- Using cached msys2-automake1.16-1.16.5-1-any.pkg.tar.zst.
-- Using cached msys2-libiconv-1.17-1-x86_64.pkg.tar.zst.
-- Using cached msys2-libintl-0.21-2-x86_64.pkg.tar.zst.
-- Using cached msys2-zlib-1.2.13-1-x86_64.pkg.tar.zst.
-- Using cached msys2-tar-1.34-3-x86_64.pkg.tar.zst.
-- Using cached msys2-findutils-4.9.0-3-x86_64.pkg.tar.zst.
-- Using cached msys2-gmp-6.2.1-2-x86_64.pkg.tar.zst.
-- Using cached msys2-gcc-libs-11.3.0-3-x86_64.pkg.tar.zst.
-- Using cached msys2-libbz2-1.0.8-4-x86_64.pkg.tar.zst.
-- Using cached msys2-liblzma-5.4.1-1-x86_64.pkg.tar.zst.
-- Using cached msys2-libzstd-1.5.4-1-x86_64.pkg.tar.zst.
-- Using cached msys2-libreadline-8.2.001-3-x86_64.pkg.tar.zst.
-- Using cached msys2-mpfr-4.2.0-2-x86_64.pkg.tar.zst.
-- Using cached msys2-libpcre-8.45-3-x86_64.pkg.tar.zst.
-- Using cached msys2-m4-1.4.19-2-x86_64.pkg.tar.zst.
-- Using cached msys2-perl-5.36.0-1-x86_64.pkg.tar.zst.
-- Using cached msys2-ncurses-6.4-1-x86_64.pkg.tar.zst.
-- Using cached msys2-libcrypt-2.1-4-x86_64.pkg.tar.zst.
-- Using msys root at G:/v/downloads/tools/msys2/d7266db249278763
-- Generating configure for x64-windows
-- Finished generating configure for x64-windows
-- Using cached mingw-w64-i686-pkgconf-1~1.8.0-2-any.pkg.tar.zst.
-- Using cached msys2-msys2-runtime-3.4.6-1-x86_64.pkg.tar.zst.
-- Using msys root at G:/v/downloads/tools/msys2/6f3fa1a12ef85a6f
-- Configuring x64-windows-dbg
-- Configuring x64-windows-rel
-- Building x64-windows-dbg
-- Installing x64-windows-dbg
-- Building x64-windows-rel
-- Installing x64-windows-rel
-- Fixing pkgconfig file: G:/v/vcpkg1/packages/glpk_x64-windows/lib/pkgconfig/glpk.pc
CMake Error at scripts/cmake/vcpkg_fixup_pkgconfig.cmake:141 (message):
  G:/v/downloads/tools/msys2/6f3fa1a12ef85a6f/mingw32/bin/pkg-config.exe
  --exists glpk failed with error code: 1

      ENV{PKG_CONFIG_PATH}: "G:/v/vcpkg1/packages/glpk_x64-windows/lib/pkgconfig;G:/v/vcpkg1/packages/glpk_x64-windows/share/pkgconfig;G:/v/vcpkg1/vcpkg_installed/x64-windows/lib/pkgconfig;G:/v/vcpkg1/vcpkg_installed/x64-windows/share/pkgconfig"
      output: Package odbc was not found in the pkg-config search path.

  Perhaps you should add the directory containing `odbc.pc'

  to the PKG_CONFIG_PATH environment variable

  Package 'odbc', required by 'glpk', not found
Call Stack (most recent call first):
  scripts/cmake/vcpkg_fixup_pkgconfig.cmake:206 (z_vcpkg_fixup_pkgconfig_check_files)
  ports/glpk/portfile.cmake:64 (vcpkg_fixup_pkgconfig)
  scripts/ports.cmake:170 (include)



```

### PR DESCRIPTION
**Pull request by LilyWangLL**
Fixed https://github.com/microsoft/vcpkg/issues/35245
The related: https://github.com/microsoft/vcpkg/issues/33751#issuecomment-1720822828
Fix `glpk[odbc]` and `glpk[mysql]` build error:

```
      output: Package odbc was not found in the pkg-config search path.
  Perhaps you should add the directory containing `odbc.pc'
  to the PKG_CONFIG_PATH environment variable
  Package 'odbc', required by 'glpk', not found
```
```
      output: Package mysql was not found in the pkg-config search path.
  Perhaps you should add the directory containing `mysql.pc'
  to the PKG_CONFIG_PATH environment variable
  Package 'mysql', required by 'glpk', not found
```

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [ ] ~~SHA512s are updated for each updated download~~
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [ ] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/glpk/portfile.cmake [hunk 1]
```diff
@@ -34,14 +34,12 @@ if("mysql" IN_LIST FEATURES)
...
         "CPPFLAGS=-I${CURRENT_INSTALLED_DIR}/include/mysql \$CPPFLAGS"
     )
-    string(APPEND requires " mysql")
 else()
     vcpkg_list(APPEND CONFIGURE_OPTIONS --disable-mysql)
 endif()
 
 if("odbc" IN_LIST FEATURES)
     vcpkg_list(APPEND CONFIGURE_OPTIONS --enable-odbc)
-    string(APPEND requires " odbc")
```
Comment (dg0yt) [inline ports/glpk/portfile.cmake:44]: So how does the pc files carry the link libraries now? Do they appear in `Libs`?
BTW `@requires@` should probably be for `Requires.private`, not for `Requires`, in glpk.pc.in.

**Review by Cheney-W — APPROVED** [review]

**Comment by dg0yt** [comment]
I guess the features' exports are really broken now.
`-lmysql` won't find our `libmysqlclient.a`.
And odbc links to what? If it is really for iodbc, than maybe it is time to add a port.

**Comment by LilyWangLL** [comment]
> I guess the features' exports are really broken now. `-lmysql` won't find our `libmysqlclient.a`. 

About this, I will change to the following codes:
```
    if(VCPKG_TARGET_IS_WINDOWS)
        string(APPEND libs " -lmysql")
    else()
        string(APPEND libs " -lmysqlclient")
    endif()
```

> And odbc links to what? If it is really for iodbc, than maybe it is time to add a port.

> iODBC table driver, which provides connection to a database through ODBC
> Under Debian GNU/Linux the ODBC table driver uses the iODBC

It seems `odbc` and `iodbc` all are needed.

**Comment by dg0yt** [comment]
> if(VCPKG_TARGET_IS_WINDOWS)
>         string(APPEND libs " -lmysql")
>     else()
>         string(APPEND libs " -lmysqlclient")
>     endif()

I suggest to not use libs but to use requires. Because static linkage probably comes with additional link libs which should already in the pc file which is installed by mysql.

**Pull request merged**